jetty-server/src/main/java/org/eclipse/jetty/server/AsyncContentProducer.java

```diff
@@ -219,39 +219,43 @@ public void reclaim(HttpInput.Content content)
     public boolean isReady()
     {
         HttpInput.Content content = nextTransformedContent();
-        if (content == null)
+        if (content != null)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("isReady(), got transformed content {} {}", content, this);
+            _httpChannel.getState().onContentAdded();
+            return true;
+        }
+
+        _httpChannel.getState().onReadUnready();
```

Comment:
So @lorban just double checking that we don't need to call `onReadUnready` in the body of this loop.... because `onContentAdded` is not called if we loop.

Reply:
`onReadUnready` has no side effect, it only changes the state to `UNREADY`.
We **have** to be in `UNREADY` state **before** the call to `needContent()` if it ever returns false, but we do not need to if it returns true. So when `needContent()` returns true, the transition to `UNREADY` was unnecessary, but harmless.
So calling `onReadUnready()` in the loop is useless, we just need to make sure we're in `UNREADY` state before we call `needContent()` and we're good to go.